lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -400,7 +399,19 @@ impl<'s> SegmentHolder {
         let mut max_persisted_version: SeqNumberType = SeqNumberType::MIN;
         let mut min_unsaved_version: SeqNumberType = SeqNumberType::MAX;
         let mut has_unsaved = false;
-        for segment in self.segments.values() {
+
+        let appendable_segments = self.appendable_segments();
```

Comment:
I propose to introduce a separate method `fn flush_ordered_segment() -> Vec<SegmentId>` with segments flush ordering